packages/storage/src/cookies.ts

```diff
@@ -52,20 +51,25 @@ function deserializeCookieOptions(cookie: string, key: string) {
   return cookie.match(`(^|;)\\s*${key}\\s*=\\s*([^;]+)`)?.pop() ?? null;
 }
 
-let useRequest: () => PageEvent | undefined;
-try {
-  useRequest = require("solid-start/server").useRequest;
-} catch (e) {
-  useRequest = () => {
-    // eslint-disable-next-line no-console
-    console.warn(
-      "It seems you attempt to use cookieStorage on the server without having solid-start installed or use vite.",
+const getRequestHeaders = isServer
+  ? () => getRequestEvent()?.request?.headers || new Headers()
+  : () => new Headers();
+const setCookie = isServer
+  ? (key: string, value: string, options: CookieProperties) => {
+    const responseHeaders = (
+      getRequestEvent() as unknown as RequestEvent & { response?: { headers: Headers } }
+    )?.response?.headers;
+    if (!responseHeaders) {
+      return;
+    }
+    const serializedOptions = serializeCookieOptions(options);
+    responseHeaders.set(
+      "Set-Cookie",
+      `${responseHeaders.get("Set-Cookie") ? ", " : ""}${key}=${value}${serializedOptions ? "; " : ""
```

Comment:
What is this `,` at the beginning for?

Reply:
If there are already cookies in the request, we do not want to overwrite them, but to append.

Reply:
No, I wanted set to clean up previous cookies for the same key. Let's do that in two steps, I guess.

Reply:
Now I only need to reconcile and cleanup request and response cookies (the latter should take precedence) and then we can proceed.

Reply:
I completely rewrote the server read/write part and even changed the API so we now get the headers for the response as well (since we need it for reading).